homeassistant/components/zha/update.py

```diff
@@ -0,0 +1,171 @@
+"""Representation of ZHA updates."""
+# pylint: skip-file
+from __future__ import annotations
+
+import asyncio
+from datetime import datetime
+import functools
+from typing import TYPE_CHECKING, Any
+
+import zigpy.exceptions
+from zigpy.ota.image import BaseOTAImage
+from zigpy.zcl.foundation import Status
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .core import discovery
+from .core.const import CLUSTER_HANDLER_OTA, SIGNAL_ADD_ENTITIES
+from .core.helpers import get_zha_data
+from .core.registries import ZHA_ENTITIES
+from .entity import ZhaEntity
+
+if TYPE_CHECKING:
+    from .core.cluster_handlers import ClusterHandler
+    from .core.device import ZHADevice
+
+CONFIG_DIAGNOSTIC_MATCH = functools.partial(
+    ZHA_ENTITIES.config_diagnostic_match, Platform.UPDATE
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Zigbee Home Automation update from config entry."""
+    zha_data = get_zha_data(hass)
+    entities_to_create = zha_data.platforms[Platform.UPDATE]
+
+    unsub = async_dispatcher_connect(
+        hass,
+        SIGNAL_ADD_ENTITIES,
+        functools.partial(
+            discovery.async_add_entities, async_add_entities, entities_to_create
+        ),
+    )
+    config_entry.async_on_unload(unsub)
+
+
+@CONFIG_DIAGNOSTIC_MATCH(cluster_handler_names=CLUSTER_HANDLER_OTA)
+class ZHAFirmwareUpdateEntity(ZhaEntity, UpdateEntity):
+    """Representation of a ZHA firmware update entity."""
+
+    _unique_id_suffix = "firmware_update"
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_device_class = UpdateDeviceClass.FIRMWARE
+    _attr_supported_features = (
+        UpdateEntityFeature.INSTALL
+        | UpdateEntityFeature.PROGRESS
+        | UpdateEntityFeature.SPECIFIC_VERSION
+    )
+
+    def __init__(
+        self,
+        unique_id: str,
+        zha_device: ZHADevice,
+        channels: list[ClusterHandler],
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the ZHA update entity."""
+        super().__init__(unique_id, zha_device, channels, **kwargs)
+        self._ota_cluster_handler: ClusterHandler = self.cluster_handlers[
+            CLUSTER_HANDLER_OTA
+        ]
+        self._attr_installed_version: str = self.zha_device.sw_version or "unknown"
+        self._latest_version_firmware: BaseOTAImage = None
+        self._attr_name = "Firmware"
```

Comment:
I think ZHA has `has_entity_name = True`, if so, you can omit the `_attr_name`, because for Update entities it will use the device class translation for Firmware if there is no name or translation key.